packages/11ty/_includes/components/objects-catalog.webc

```diff
@@ -0,0 +1,588 @@
+<template webc:nokeep>
+  <objects-data :objects="objects"></objects-data>
+  <section class="object-filters">
+    <input class="collapsible__checkbox" type="checkbox" id="collapsible-checkbox" />
+    <label for="collapsible-checkbox" class="collapsible__toggle"></label>
+    <div class="collapsible__panel">
+      <object-filters :filters="filters" :objects="objects"></object-filters>
+    </div>
+    <div class="collapsible__controls">
+      <div class="object-filters__controls">
+        <button class="object-filters__button object-filters__button--primary" data-control-apply>Apply Filters</button>
+        <button class="object-filters__button" data-control-reset>Reset</button>
+      </div>
+    </div>
+  </section>
+  <section class="object-cards">
+    <object-cards :objects="objects"></object-cards>
+  </section>
+</template>
+<script webc:setup>
+const { filePathStem } = this.page;
+const currentDir = filePathStem.split('/').slice(0, -1).join('/');
+const collection = this.collections.tableOfContentsHtml;
+
+// We want to find the index page in the current directory
+const currentDirIndexPage = this
+  .eleventyNavigation(collection)
+  .find(({ key }) => key === this.slugify(currentDir));
+
+// 1. we want to get the current directory's index page's children.
+// 2. of the children, all pages that have an `object` key in their `data` are relevant
+// 3. page objects can either have an id referencing an object in objects.yaml, or the full object data in front matter
+// 4. if the current page has children, it is an index page; return to step 1 to retrieve all objects in the new current directory
+
+const objects = [];
+
+const addObject = ({ children, data, url }) => {
+  if (children && children.length) {
+    children.forEach(addObject);
+  }
+  /**
+   * Renaming `object` to the pluralized `pageObjects` to clarify that we want
+   * ALL objects listed in front matter (even if the page only currently renders
+   * the first one), and so we can more clearly refer to individual objects
+   */
+  const { object: pageObjects } = data;
+  if (!pageObjects) return;
+  pageObjects.forEach((object) => {
+    const objectData = object.id ? this.getObject(object.id) : object;
+    const thumbnail = objectData.figures
+      ? objectData.figures.find(({ mediaType }) => mediaType === 'image').thumbnail
+      : objectData.thumbnail;
+    console.warn('THUMBNAIL?', thumbnail);
+    /**
+     * Passing `url` along with object data to provide links to the object page
+     */
+    objects.push({ ...objectData, thumbnail, url });
+  });
+}
+
+currentDirIndexPage.children.forEach(addObject)
```

Comment:
In this section, we are gathering up the `objects` data to render as JSON. This has been refactored to recursively check all subdirectories for objects in the current dir